File: lib/model/ware.dart
import 'dart:convert';

class Ware {
  final String wareName;
  final String unit;
  final String group;
  final String description;
  final num cost;
  final num sale;
  final num quantity;
  final String? id;

  Ware({
    required this.wareName,
    required this.unit,
    required this.group,
    required this.description,
    required this.cost,
    required this.sale,
    required this.quantity,
    this.id,
  });



  Map<String, dynamic> toMap() {
    return {
      'wareName': wareName,
      'unit': unit,
      'group': group,
      'description':description,
      'cost': cost,
      'sale': sale,
      'quantity': quantity,
      'id': id,
    };
  }

  factory Ware.fromMap(Map<String, dynamic> map) {
    return Ware(
      wareName: map['wareName'] ?? "",
      unit: map['unit'] ?? "",
      group: map['group'] ?? "",
      description: map['description'] ?? "",
      cost: map['cost'] ?? 0,
      sale: map['sale'] ?? 0,
      quantity: map['quantity'] ?? 0,
      id: map['id'] ?? "",
    );
  }
  String toJson()=>json.encode(toMap());
  factory Ware.fromJson(String source) => Ware.fromMap(json.decode(source),);
}

File: server/routes/ware.js
const express = require("express");
const crypto = require("crypto");
const wareRouter = express.Router();
const ware = require("../middlewars/ware");
const saver = require("../saver/saver.js");

//Post ware on db
wareRouter.post("/user/ware/post", ware, async (req, res) => {
    try {
        var { wareName, unit, group, description, cost, sale, quantity, id } = req.body;

       var _id=crypto.randomBytes(16).toString("hex");
        await saver.saveToJson(
            {wareName, unit, group, description, cost, sale, quantity, id:_id },
             "./db/user/wares.json");

        res.json({ id });
    } catch (e) {
        res.status(500).json({ "error": e });
        console.log(e);
    }
});

//read wares from db
wareRouter.get("/user/ware/get-wares", async (req, res) => {
    var waresList = saver.readFromJson("./db/user/wares.json");
    res.json(waresList);

});

//delete ware 
wareRouter.post("/user/ware/delete-ware",async (req,res)=>{
try{
    const {id}=req.body;
saver.deleteFromJson("./db/user/wares.json",id);
res.json({msg:"deleted Successfully"});
}catch(e){
res.status(500).json({error:e});
console.log(e);
}
});

//upadte ware
wareRouter.post("/user/ware/update-ware",async (req,res)=>{
    try{
        const {wareName,unit,sale,cost,group,description,quantity,id}=req.body;
    saver.updateJsonObject("./db/user/wares.json",id,{wareName,unit,sale,cost,group,description,quantity,id});
    res.json({msg:"update Successfully"});
    }catch(e){
    res.status(500).json({error:e});
    console.log(e);
    }
    });

module.exports = wareRouter;